patch_test.go

```diff
@@ -0,0 +1,84 @@
+package quicktest_test
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+)
+
+func TestSetInt(t *testing.T) {
+	c := qt.New(t)
+	i := 99
+	c.Patch(&i, 88)
+	c.Assert(i, qt.Equals, 88)
+	c.Cleanup()
+	c.Assert(i, qt.Equals, 99)
+}
+
+func TestSetError(t *testing.T) {
+	c := qt.New(t)
+	oldErr := errors.New("foo")
+	newErr := errors.New("bar")
+	err := oldErr
+	c.Patch(&err, newErr)
+	c.Assert(err, qt.Equals, newErr)
+	c.Cleanup()
+	c.Assert(err, qt.Equals, oldErr)
+}
+
+func TestSetErrorToNil(t *testing.T) {
+	c := qt.New(t)
+	oldErr := errors.New("foo")
+	err := oldErr
+	c.Patch(&err, nil)
+	c.Assert(err, qt.Equals, nil)
+	c.Cleanup()
+	c.Assert(err, qt.Equals, oldErr)
+}
+
+func TestSetMapToNil(t *testing.T) {
+	c := qt.New(t)
+	oldMap := map[string]int{"foo": 1234}
+	m := oldMap
+	c.Patch(&m, nil)
+	c.Assert(m, qt.IsNil)
+	c.Cleanup()
+	c.Assert(m, qt.DeepEquals, oldMap)
+}
+
+func TestSetPanicsWhenNotAssignable(t *testing.T) {
+	c := qt.New(t)
+	i := 99
+	type otherInt int
+	c.Assert(func() { c.Patch(&i, otherInt(88)) }, qt.PanicMatches, `reflect\.Set: value of type quicktest_test\.otherInt is not assignable to type int`)
+}
+
+func TestSetenv(t *testing.T) {
+	c := qt.New(t)
+	const envName = "TESTING_PATCH_ENVIRONMENT"
+	// remember the old value, and set it to something we can check
+	os.Setenv(envName, "initial")
+	c.Setenv(envName, "new value")
+	// Using check to make sure the environment gets set back properly in the test.
+	c.Check(os.Getenv(envName), qt.Equals, "new value")
+	c.Cleanup()
+	c.Check(os.Getenv(envName), qt.Equals, "initial")
+}
+
+func TestMkdir(t *testing.T) {
+	c := qt.New(t)
+	dir := c.Mkdir()
+	c.Assert(c, qt.Not(qt.Equals), "")
+	info, err := os.Stat(dir)
+	c.Assert(err, qt.Equals, nil)
+	c.Assert(info.IsDir(), qt.Equals, true)
+	f, err := os.Create(filepath.Join(dir, "hello"))
+	c.Assert(err, qt.Equals, nil)
+	f.Close()
+	c.Cleanup()
+	_, err = os.Stat(dir)
+	c.Assert(err, qt.Not(qt.IsNil))
```

Comment:
`c.Assert(os.IsNotExist(err), qt.Equals, true)` ?

Reply:
Yeah, maybe. I miss Satisfies - maybe we could add that.